contrib/autopilot/c-lightning-autopilot.py

```diff
@@ -0,0 +1,269 @@
+            res.append("".join(
+                ['{:1x}'.format(integer) for integer in decoded])[:-1])
+        return res
+            
+    
+    def __connect_to_seeds(self):
+        """
+        sets up peering connection to seed nodes of the lightning network
+        
+        This is necessary in case the node operating the autopilot has never
+        been connected to the lightning network.
+        """
+        try:
+            for nodeid in random.shuffle(self.__get_seed_keys()):
+                self.__clogger.info("peering with node: " + nodeid)
+                self.__rpc_interface.connect(nodeid)
+                # FIXME: better strategy than sleep(2) for building up
+                time.sleep(2)
+        except:
+            pass
+    
+    def __download_graph(self):
+        """
+        Downloads a local copy of the nodes view of the lightning network
+        
+        This copy is retrieved by listnodes and listedges RPC calls and will
+        thus be incomplete as peering might not be ready yet.
+        """
+        
+        # FIXME: it is a real problem that we don't know how many nodes there
+        # could be. In particular billion nodes networks will outgrow memory
+        G = nx.Graph()
+        self.__clogger.info("Instantiated networkx graph to store the lightning network")
+        
+        nodes = []
+        try:
+            self.__clogger.info(
+                "Attempt RPC-call to download nodes from the lightning network")
+            while len(nodes) == 0:
+                peers = self.__rpc_interface.listpeers()["peers"]
+                if len(peers) < 1:
+                    self.__connect_to_seeds()
+                nodes = self.__rpc_interface.listnodes()["nodes"]
+        except ValueError as e:
+            self.__clogger.info(
+                "Node list could not be retrieved from the peers of the lightning network")
+            self.__clogger.debug("RPC error: " + str(e))
+            raise e 
+
+        for node in nodes:
+            G.add_node(node["nodeid"], **node)
+
+        self.__clogger.info(
+            "Number of nodes found and added to the local networkx graph: {}".format(len(nodes)))
+
+            
+        channels = {}
+        try:
+            self.__clogger.info(
+                "Attempt RPC-call to download channels from the lightning network")
+            channels = self.__rpc_interface.listchannels()["channels"]
+            self.__clogger.info(
+                "Number of retrieved channels: {}".format(
+                    len(channels)))
+        except ValueError as e:
+            self.__clogger.info(
+                "Channel list could not be retrieved from the peers of the lightning network")
+            self.__clogger.debug("RPC error: " + str(e))
+            return False
+                
+        for channel in channels:
+            G.add_edge(
+                channel["source"],
+                channel["destination"],
+                **channel)
```

Comment:
> I have thought about an explicit mapping but decided to take the c-lightning names as the standard - since currently the lib is only implemented in c-lightning. if someone else wants to use it they need to define the mapping in their wrapper. still an API for it in the lib autopilot seems reasonable. Not sure if this should be done in the first version. If I provide those there would also be the need for tests to fix the API